crates/types/src/config/bifrost.rs

```diff
@@ -318,16 +323,25 @@ pub struct ReplicatedLogletOptions {
     pub default_nodeset_size: NodeSetSize,
 }
 
-fn nodeset_size_is_zero(i: &NodeSetSize) -> bool {
-    *i == NodeSetSize::ZERO
+impl ReplicatedLogletOptions {
+    fn apply_common(&mut self, _common: &CommonOptions) {
+        #[allow(deprecated)]
+        self.default_log_replication.as_ref().inspect(|_| {
+            print_warning_deprecated_config_option(
+                "bifrost.replicated-loglet.default-log-replication",
+                Some("default-replication"),
+            );
+        });
+    }
```

Comment:
I get why you've done it like this but I am not a huge fan of using `apply_common` to print deprecation warnings. I think it's a tad bit cleaner how we've done it with other option structs where we have *Shadow which handles deprecation etc. and can also be used for printing warnings. That way, `ReplicatedLogletOptions` wouldn't have to keep `default_log_replication`. On the other hand, with a *Shadow, we need to duplicate `ReplicatedLogletOptions`.